lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -222,9 +248,9 @@ def _should_request_new_certificate(self) -> bool:
         pending_csrs = self.tls_access.get_certificate_signing_requests(unfulfilled_only=True)
         expired_certs = self.tls_access.get_expiring_certificates()
 
-        existing_csr_is_fulfilled = any([existing_csr in csr.values() for csr in fulfilled_csrs])
-        existing_csr_is_pending = any([existing_csr in csr.values() for csr in pending_csrs])
-        existing_csr_expiring = any([existing_csr in cert.values() for cert in expired_certs])
+        existing_csr_is_fulfilled = any([existing_csr in csr.csr for csr in fulfilled_csrs])
+        existing_csr_is_pending = any([existing_csr in csr.csr for csr in pending_csrs])
+        existing_csr_expiring = any([existing_csr in cert.certificate for cert in expired_certs])
```

Comment:
`csr.csr` reads weird here. maybe replace replace the name of the `csr` object with `object` or `entry` or something like that? It's not that important though.

Reply:
done